Lib/asyncio/base_events.py

```diff
@@ -1306,8 +1306,12 @@ def _check_sendfile_params(self, sock, file, offset, count):
 
             exceptions = []
 
-            if reuse_address is None:
-                reuse_address = os.name == 'posix' and sys.platform != 'cygwin'
+            if reuse_address:
+                # bpo-37228
+                raise RuntimeError("Passing `reuse_address=True` is no "
```

Comment:
I'm not entirely certain as to whether this would be more appropriate as a `RuntimeError` or `ValueError`. At
first I was considering `ValueError` due to it only occurring when passing `reuse_address=True`, but the
error itself is more related to `reuse_address` parameter no longer being supported, rather than just being passed an invalid value:

```
ValueError - Raised when an operation or function receives an argument that has the right type but an
inappropriate value, and the situation is not described by a more precise exception such as IndexError.
```

AFAIK, there's not a more specific exception class that would be suitable for this situation. As a result, I think the appropriate choice would be `RuntimeError`:

```
RuntimeError - Raised when an error is detected that doesn’t fall in any of the other categories. The
associated value is a string indicating what precisely went wrong.
```

I'm definitely open to suggestions on this though, including other exception classes.

Reply:
Since @pitrou initially proposed the idea for explicitly passing `reuse_address=True` to raise an error, I'm curious as to what his thoughts are on this.

Reply:
This should be `ValueError`. It's not a runtime condition, and it won't vary depending on external conditions.

Reply:
@pitrou 
> It's not a runtime condition, and it won't vary depending on external conditions.

Ah, I see. Perhaps I was using `RuntimeError` as a bit too much of a "catch all" based on it's description in the [documentation](https://docs.python.org/3.8/library/exceptions.html#RuntimeError).

Do you think the docs for `RuntimeError`  should explicitly mention that it should be primarily used for exceptions that vary based on external conditions?

Either way, I've updated the PR accordingly. Passing `reuse_address=True` will now raise a `ValueError` instead of a `RuntimeError`.